.github/workflows/ci.yml

```diff
@@ -287,28 +381,61 @@ jobs:
           echo $(git rev-parse HEAD) > canary-latest.txt
           gsutil cp canary-latest.txt gs://dl.deno.land/canary-latest.txt
 
-      - name: Test release
-        if: matrix.kind == 'test' && matrix.profile == 'release'
-        run: cargo test --release --locked --all-targets
-
       - name: Test debug
-        if: matrix.kind == 'test' && matrix.profile == 'debug'
+        if: |
+          matrix.kind == 'test' && matrix.profile == 'debug' &&
+          !matrix.use_sysroot
         run: |
           cargo test --locked --doc
-          cargo test --locked --all-targets
+          cargo test --locked
 
-      # TODO(ry) Because CI is so slow on for OSX and Windows, we currently only run WPT on Linux.
-      - name: Configure hosts file for WPT (linux)
-        if: startsWith(matrix.os, 'ubuntu')
+      - name: Test release
+        if: |
+          matrix.kind == 'test' && matrix.profile == 'release' &&
+          !matrix.use_sysroot
+        run: cargo test --release --locked
+
+      - name: Test release (in sysroot)
+        if: |
+          matrix.kind == 'test' && matrix.profile == 'release' &&
+          matrix.use_sysroot
+        run: |
+          sudo chroot /sysroot   \
+               su -l "$(whoami)" \
+                  -c "cargo test --release --locked"
+
+      # TODO(ry): Because CI is so slow on for OSX and Windows, we currently
+      #           run the Web Platform tests only on Linux.
+      - name: Configure hosts file for WPT
+        if: startsWith(matrix.os, 'ubuntu') && matrix.kind == 'test'
         run: ./wpt make-hosts-file | sudo tee -a /etc/hosts
         working-directory: test_util/wpt/
 
+      - name: Run web platform tests (debug)
+        if: |
+          startsWith(matrix.os, 'ubuntu') && matrix.kind == 'test' &&
+          matrix.profile == 'debug'
+        run: |
+          ./target/debug/deno run --allow-env --allow-net --allow-read \
+                                  --allow-run --allow-write --unstable \
+                              ./tools/wpt.ts setup
+          ./target/debug/deno run --allow-env --allow-net --allow-read \
+                                  --allow-run --allow-write --unstable \
+                              ./tools/wpt.ts run --binary --quiet
```

Comment:
You will need to actually specify a binary (`./target/debug/deno `)

Reply:
A bit silly to specify `./foo/bar/deno run blah.js --binary ./foo/bar/deno`, no?

Reply:
In many scenarios the binary you want to test is not the same one that is coordinating the test.

Reply:
The --binary option does not work in release mode.